### PR TITLE
Role mapping

### PR DESCRIPTION
**Pull request by pierrotsmnrd**
This PR aims at solving issue #491

### CODE REVIEW

**Review by costrouc — CHANGES_REQUESTED** [review]
Overall looks good to me @pierrotsmnrd. I will try and run the tests that are failing and see what it is.

File: conda-store-server/conda_store_server/orm.py [hunk 1]
```diff
@@ -51,6 +51,26 @@ class Namespace(Base):
...
+class NamespaceRoleMapping(Base):
+    """Mapping between roles and namespaces"""
+
+    __tablename__ = "namespace_role_mapping"
+
+    id = Column(Integer, primary_key=True)
+    namespace_id = Column(Integer, ForeignKey("namespace.id"), nullable=False)
+    namespace = relationship(Namespace, back_populates="roles_mappings")
+
+    # arn e.g. <namespace>/<name> like `quansight-*/*` or `quansight-devops/*`
```
[the remark this answers is not in the document]
Comment (pierrotsmnrd) [reply conda-store-server/conda_store_server/orm.py:72]: I have added a validation and added some tests for this : 
https://github.com/Quansight/conda-store/pull/496/commits/d31ceaad1ebdd417f504e1fb4c0b0f63a2b3d4a7

File: conda-store-server/conda_store_server/alembic/versions/b387747ca9b7_role_mapping.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        sa.Column("role", sa.Unicode(length=255), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["namespace_id"],
+            ["namespace.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+
+    op.add_column(
+        "namespace", sa.Column("role_mapping_metadata", sa.JSON(), nullable=True)
```
[the remark this answers is not in the document]
Comment (pierrotsmnrd) [reply conda-store-server/conda_store_server/alembic/versions/b387747ca9b7_role_mapping.py:34]: `metadata_` works. `meta_data` as well. or anything like `extra_data`, `external_data`, `custom_data`

File: conda-store-server/conda_store_server/server/auth.py [hunk 3]
```diff
@@ -232,13 +233,34 @@ def is_subset_entity_permissions(
...
+                                                FROM namespace n
+                                                RIGHT JOIN namespace_role_mapping nrm ON nrm.namespace_id = n.id
+                                                WHERE n.name = :primary_namespace
+                                                """
+            ),
+            {"primary_namespace": entity.primary_namespace},
+        )
+        raw_role_mappings = result.mappings().all()
+
+        db_role_mappings = {}
```
[the remark this answers is not in the document]
Comment (pierrotsmnrd) [reply conda-store-server/conda_store_server/server/auth.py]: Haha yes, you're right. Some habits/patterns are hard to get rid of :) Done.

**Review by costrouc — APPROVED** [review]

**Pull request merged**